roles/common/DESIGN.md

```diff
@@ -0,0 +1,49 @@
+# Design Description for Common Role
+
+## Let's Encrypt Support
```

Comment:
Not everyone will be familiar with Let's encrypt, we need a short introduction ("Let's Encrypt is an automated certificate authority providing SSL certificates that are trusted by all major browsers").